### PR TITLE
add trailingComponent prop to CTABanner component

### PR DESCRIPTION
**Pull request by joshfarrant**
## Summary

Add `trailingComponent` prop to CTABanner as an escape hatch to render arbitrary content.

## List of notable changes:

- CTABanner component accepts a `trailingComponent` prop 

## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/3199

## What should reviewers focus on?

- Check that spacing is correct around trailing component.

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

image

image

### CODE REVIEW

File: packages/react/src/CTABanner/CTABanner.visual.spec.ts [hunk 1]
```diff
@@ -43,6 +43,15 @@ test.describe('Visual Comparison: CTABanner', () => {
     expect(await page.screenshot()).toMatchSnapshot()
   })
 
+  test('CTABanner / With Trailing Component', async ({page}) => {
```
Comment (rezrah) [inline packages/react/src/CTABanner/CTABanner.visual.spec.ts:46]: Can't see the new snapshot image in this PR. Could you try running it again please?

File: packages/react/src/CTABanner/CTABanner.tsx [hunk 2]
```diff
@@ -68,6 +76,7 @@ const Root = forwardRef(
         >
           <div className={clsx(styles['CTABanner-content'], align === 'center' && styles['CTABanner-content--center'])}>
             {children}
+            {TrailingComponent && <TrailingComponent />}
```
Comment (rezrah) [inline packages/react/src/CTABanner/CTABanner.tsx:79]: @joshfarrant How does this look with `align="center"`? Could you share a screenie please?

Comment (joshfarrant) [reply packages/react/src/CTABanner/CTABanner.tsx:79]: ofc, just added it to PR description

File: packages/react/src/CTABanner/CTABanner.visual.spec.ts (hunk 1 above)
Comment (joshfarrant) [reply packages/react/src/CTABanner/CTABanner.visual.spec.ts]: Good spot 👍 Looks like they've generated now, although I'm not sure why this has changed

`packages/react/src/Tooltip/Tooltip.visual.spec.ts-snapshots/Visual-Comparison-Tooltip-Tooltip-Label-Type-1-linux.png`

Comment (joshfarrant) [reply packages/react/src/CTABanner/CTABanner.visual.spec.ts]: Ah nvm, looks like they just weren't regenerated after this commit 

https://github.com/primer/brand/pull/547/commits/015bfea49cd33fd3065dfc4c7e7b742b2a45647d#diff-7e60c94a63a2596dd7d646b8d1bbd5bc87baf66db760adc47f70150c93b842e5

Looks fine to me

**Pull request merged**